### PR TITLE
Change Reports Table Display

### PR DESCRIPTION
**Pull request by davidcui1225**
*Issue #, if available:*

*Description of changes:*
* `Report ID` column now displays `<ReportSourceType>_<UUID>`, e.g `Dashboard_abcdefghijk123`
* `description` column in schema now contains the Report source name to be displayed in the `Report source` field 
* Search bar in `Reports table` now prompts user to search by `Source` in addition to Report ID
* Updated snapshot tests 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zhongnansu** [comment]
Is this 7.9.1 specific feature? Do we need to push any changes to `dev`?

**Comment by davidcui1225** [comment]
> Is this 7.9.1 specific feature? Do we need to push any changes to `dev`?

No, this issue is not present in `dev`

File: kibana-reports/public/components/main/main_utils.tsx [hunk 1]
```diff
@@ -59,9 +59,7 @@ export const addReportsTableContent = (data) => {
       sender: `\u2014`,
       kibanaRecipients: `\u2014`,
       emailRecipients: `\u2014`,
-      reportSource: reportParams.report_name.substring(
-        0, reportParams.report_name.lastIndexOf('_') 
-      ), // remove uuid from report name to display report source
+      reportSource: reportParams.description, // display report source name
```
Comment (zhongnansu) [inline kibana-reports/public/components/main/main_utils.tsx:62]: Is this a typo? We are assigning description to the source field

**Review by zhongnansu — APPROVED** [review]

File: kibana-reports/public/components/main/main_utils.tsx (hunk 1 above)
Comment (davidcui1225) [reply kibana-reports/public/components/main/main_utils.tsx:62]: Not a typo- we need a place to store the raw report source name without parsing out any special characters. And the description field is not used anywhere else.

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**